src/fastpm.c

```diff
@@ -1115,7 +1128,10 @@ smesh_force_handler(FastPMSolver * fastpm, FastPMForceEvent * event, struct smes
         FastPMFloat * delta_k = pm_alloc(fastpm->basepm);
 
         /* generate linear field at this time */
-        prepare_deltak(fastpm, fastpm->basepm, delta_k, prr, event->a_f, CONF(prr->lua, read_lineark), CONF(prr->lua, read_powerspectrum));
+        prepare_deltak(fastpm, fastpm->basepm, delta_k, prr, event->a_f, 
```

Comment:
Nice you caught this. However we will remove all of the smesh related code as it never worked and a correct implementation is too expensive in both running time and coding time.